test/src/meshgenerators/AlwaysDeleteRemotesGeneratedMeshGenerator.C

```diff
@@ -27,89 +27,13 @@ registerMooseObject("MooseTestApp", AlwaysDeleteRemotesGeneratedMeshGenerator);
-
-  params.addParam<bool>(
-      "gauss_lobatto_grid",
-      false,
-      "Grade mesh into boundaries according to Gauss-Lobatto quadrature spacing.");
-  params.addRangeCheckedParam<Real>(
-      "bias_x",
-      1.,
-      "bias_x>=0.5 & bias_x<=2",
-      "The amount by which to grow (or shrink) the cells in the x-direction.");
-  params.addRangeCheckedParam<Real>(
-      "bias_y",
-      1.,
-      "bias_y>=0.5 & bias_y<=2",
-      "The amount by which to grow (or shrink) the cells in the y-direction.");
-  params.addRangeCheckedParam<Real>(
-      "bias_z",
-      1.,
-      "bias_z>=0.5 & bias_z<=2",
-      "The amount by which to grow (or shrink) the cells in the z-direction.");
-
-  params.addParam<std::string>("boundary_name_prefix",
-                               "If provided, prefix the built in boundary names with this string");
-  params.addParam<boundary_id_type>(
-      "boundary_id_offset", 0, "This offset is added to the generated boundary IDs");
-
-  params.addParam<std::vector<ExtraElementIDName>>("extra_element_integers",
-                                                   "Names of extra element integers");
-
-  params.addClassDescription(
-      "Create a line, square, or cube mesh with uniformly spaced or biased elements.");
-
-  return params;
+  return GeneratedMeshGenerator::validParams();
 }
 
 AlwaysDeleteRemotesGeneratedMeshGenerator::AlwaysDeleteRemotesGeneratedMeshGenerator(
     const InputParameters & parameters)
-  : MeshGenerator(parameters),
-    _dim(getParam<MooseEnum>("dim")),
-    _nx(declareMeshProperty("num_elements_x", getParam<unsigned int>("nx"))),
-    _ny(declareMeshProperty("num_elements_y", getParam<unsigned int>("ny"))),
-    _nz(declareMeshProperty("num_elements_z", getParam<unsigned int>("nz"))),
-    _xmin(declareMeshProperty("xmin", getParam<Real>("xmin"))),
-    _xmax(declareMeshProperty("xmax", getParam<Real>("xmax"))),
-    _ymin(declareMeshProperty("ymin", getParam<Real>("ymin"))),
-    _ymax(declareMeshProperty("ymax", getParam<Real>("ymax"))),
-    _zmin(declareMeshProperty("zmin", getParam<Real>("zmin"))),
-    _zmax(declareMeshProperty("zmax", getParam<Real>("zmax"))),
-    _has_subdomain_ids(isParamValid("subdomain_ids")),
-    _gauss_lobatto_grid(getParam<bool>("gauss_lobatto_grid")),
-    _bias_x(getParam<Real>("bias_x")),
-    _bias_y(getParam<Real>("bias_y")),
-    _bias_z(getParam<Real>("bias_z")),
-    _boundary_name_prefix(isParamValid("boundary_name_prefix")
-                              ? getParam<std::string>("boundary_name_prefix") + "_"
-                              : ""),
-    _boundary_id_offset(getParam<boundary_id_type>("boundary_id_offset"))
+  : GeneratedMeshGenerator(parameters)
 {
-  if (_gauss_lobatto_grid && (_bias_x != 1.0 || _bias_y != 1.0 || _bias_z != 1.0))
-    mooseError("Cannot apply both Gauss-Lobatto mesh grading and biasing at the same time.");
 }
 
 std::unique_ptr<MeshBase>
```

Comment:
could you just call generate from the GMG too and add the one line after?

Reply:
You have to insert `allow_remote_element_removal(false)` in between `builMeshBaseObject` and `build_square/cube`

Reply:
Ok good for you for pushing on this more. Most recent commit I was able to reuse a lot more

Reply:
nice